server/channels/store/sqlstore/post_store.go

```diff
@@ -1043,14 +1056,30 @@ func (s *SqlPostStore) PermanentDeleteByChannel(channelId string) (err error) {
 	}
 	defer finalizeTransactionX(transaction, &err)
 
-	results := []postIds{}
-	err = transaction.Select(&results, "SELECT Id, RootId, UserId FROM Posts WHERE ChannelId = ?", channelId)
+	ids := []string{}
+	err = transaction.Select(&ids, "SELECT Id FROM Posts WHERE ChannelId = ?", channelId)
 	if err != nil {
 		return errors.Wrapf(err, "failed to fetch Posts with channelId=%s", channelId)
 	}
 
-	for _, ids := range results {
-		if err = s.permanentDeleteThreads(transaction, ids.Id); err != nil {
+	batchIds := [][]string{}
+	batchSize := 1000
```

Comment:
I decided to break this out into batches of 1000 instead of deleting each record 1 by 1. Not sure why `PermanentDeleteByUser` maxes out at 10000 posts but we allow this to delete everything?